### PR TITLE
[Lua] Improve auto-generated test files

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Based on the feedback by @daurnimator in https://github.com/swagger-api/swagger-codegen/pull/7562

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/lua/api_test.mustache [hunk 1]
```diff
@@ -18,6 +7,13 @@ Please update as you see appropriate
 {{#operations}}
 describe("{{classname}}", function()
 {{#operation}}
+  local petstore_{{classname}} = require "{{packageName}}.api.{{{classname}}}"
```
Comment (daurnimator) [inline modules/swagger-codegen/src/main/resources/lua/api_test.mustache]: Petstore here

File: samples/client/petstore/lua/spec/api_response_spec.lua [hunk 2]
```diff
@@ -1,23 +1,21 @@
 --[[
-Swagger Petstore
```
Comment (daurnimator) [inline samples/client/petstore/lua/spec/api_response_spec.lua]: Was indentation change on purpose?

File: samples/client/petstore/lua/spec/pet_api_spec.lua [hunk 3]
```diff
@@ -1,72 +1,76 @@
...
 describe("pet_api", function()
+  local petstore_pet_api = require "petstore.api.pet_api"
   -- unit tests for add_pet
   describe("add_pet test", function()
     it("should work", function()
       -- TODO assertion here: http://olivinelabs.com/busted/#asserts
     end)
   end)
 
+  local petstore_pet_api = require "petstore.api.pet_api"
```
Comment (daurnimator) [inline samples/client/petstore/lua/spec/pet_api_spec.lua]: Only need the require once at the top of describe, not before each test

File: samples/client/petstore/lua/spec/api_response_spec.lua (hunk 2 above)
Comment (wing328) [reply samples/client/petstore/lua/spec/api_response_spec.lua]: Yup, I've not manually deleted the spec files as spec files won't be overwritten during code generation.

**Pull request merged**